src/main/java/com/programmers/librarymanagement/application/LibraryManagementService.java

```diff
@@ -0,0 +1,127 @@
+    }
+
+    public void addBook(String title, String author, int page) {
+
+        Book book = new Book(bookRepository.createId(), title, author, page, Status.CAN_RENT, LocalDateTime.now());
+        bookRepository.addBook(book);
+    }
+
+    public List<Book> getAllBooks() {
+
+        updateAllStatus();
+
+        return bookRepository.findAll();
+    }
+
+    public List<Book> getBookByTitle(String title) {
+
+        updateAllStatus();
+
+        return bookRepository.findByTitle(title);
+    }
+
+    public void rentBook(Long id) {
+
+        Book book = bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+
+        // 도서가 대여 가능할 경우, 대여중으로 상태 변경
+        switch (book.getStatus()) {
+            case CAN_RENT -> {
+                book.updateStatusToAlreadyRent();
+                bookRepository.updateBook(book);
+            }
+
+            case ALREADY_RENT -> throw new BookAlreadyRentException();
+
+            case ARRANGE -> {
+                if (book.getReturnDateTime().plusMinutes(BOOK_ARRANGE_TIME).isBefore(LocalDateTime.now())) {
+                    book.updateStatusToAlreadyRent();
+                    bookRepository.updateBook(book);
+                } else {
+                    throw new BookArrangeException();
+                }
+            }
+
+            case LOST -> throw new BookLostException();
+        }
+    }
+
+    public void returnBook(Long id) {
+
+        Book book = bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+
+        // 도서가 반납 가능할 경우, 정리중으로 상태 변경
+        switch (book.getStatus()) {
+            case CAN_RENT -> throw new BookAlreadyReturnException();
+
+            case ALREADY_RENT, LOST -> {
+                book.updateStatusToArrange();
+                book.updateReturnDateTime();
+                bookRepository.updateBook(book);
+            }
+
+            case ARRANGE -> {
+
+                // 도서 반납 후 5분이 지났다면 대여 가능
+                if (book.getReturnDateTime().plusMinutes(BOOK_ARRANGE_TIME).isBefore(LocalDateTime.now())) {
+                    throw new BookAlreadyReturnException();
+                } else {
+                    throw new BookArrangeException();
+                }
+            }
+        }
+    }
+
+    public void lostBook(Long id) {
+
+        Book book = bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+
+        // 도서가 분실 처리 가능할 경우, 분실됨으로 상태 변경
+        switch (book.getStatus()) {
+            case CAN_RENT, ALREADY_RENT, ARRANGE -> {
+                book.updateStatusToLost();
+                bookRepository.updateBook(book);
+            }
+
+            case LOST -> throw new BookAlreadyLostException();
+        }
+    }
+
+    public void deleteBook(Long id) {
+
+        Book book = bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+        bookRepository.deleteBook(book);
+    }
+
+    private void updateAllStatus() {
+
+        List<Book> bookList = bookRepository.findAll();
```

Comment:
자료구조에 용이하게 변수명을 쓰기 위해서 books로 바꾸는 건 어떨까요?
```suggestion
        List<Book> books = bookRepository.findAll();
```

Reply:
자료구조가 List라서 bookList라는 변수명을 사용한 것은 아닙니다! '목록'이라는 의미를 생각하고 붙인 이름입니다
하지만 사람에 따라 List가 자료구조의 의미로 생각될 수 있을 것 같아 다음부터는 조금더 유연한 네이밍을 할 수 있도록 주의하겠습니다